### ISSUE TITLE
Code intelligence doesn't work for some AS3/MXML projects in v0.23

### ISSUE THREAD

**Issue by yooni126** [BowlerHatLLC/vscode-as3mxml#414]
Hi josh
I have a weird issue and I really don’t know how can I fix it
yesterday when I run VsCode in my mac os(Mojave) 
I saw the autoComplete codes method didn't work for me even for recent projects that I have worked with them

in fact, Vscode looks like a notepad now! 

these ways I tried to fix my problem but it didn't work:
1-check asconfig.json
2-update java(8.0.221)
3-remove VsCode cache folder from Mac library
4-download Vscode again

this is my asconfig.json:
```
{
	"animateOptions": {
		"file": "./Tel.fla"
	},
	"files": [
		"./Tel.as"
	],
	"compilerOptions": {
		"external-library-path": [
			"/Users/Younes/Documents/GitHub/MteamGoogleMapAPI/src",
			"/Users/Younes/Documents/GitHub/FlexLibs-andOthers"
		],
		"source-path": [
			"/Users/Younes/Documents/GitHub/MteamGoogleMapAPI/src",
			"/Users/Younes/Documents/GitHub/ASPack",
			"/Users/Younes/Documents/GitHub/SaffronCode",
			"/Users/Younes/Documents/GitHub/FlexLibs-andOthers"
		]
	},
	"type": "app",
	"config": "airmobile"
}
```
example from output panel and my issue problem :
https://drive.google.com/open?id=1EvSrpOuXEuGNuiz1b-X2TDUaEoagI8V4

for more information, from yesterday I was trying to run VsCode, the CPU is warm-up and Fan starts working. this is my activity monitor screenshot
https://drive.google.com/open?id=1x7BVP1SJeiAYrzofyVkRwFw96Ge1Mjtb

also, this problem only occurs when I work with ActionScript codes because when I create  *.js files the autoComplete method works without any problem

please help me to solve this crazy issue!

**Comment by yooni126** [comment]
update **
 I think this is a global error!
I tested this issue in another system with Windows 10, and this problem occurred again

**Comment by joshtynjala** [comment]
It is a bug. I'm working on finding a fix!

**Comment by joshtynjala** [comment]
@yooni126 I just published v0.23.1 with some bug fixes to the VSCode Marketplace. Can you give that new version a try and let me know if you still see the same problem? Thanks!

**Comment by yooni126** [comment]
Thank you josh 
but the problem didn't fix 
after 3 days and plenty of reviews between many classes, I found a line code that makes this problem
when I comment it, everything like CPU becomes works normally

this is a fucking code:
`for each(var product:StoreKitProduct in e.validProducts) `

**Comment by joshtynjala** [comment]
Thank you. I can confirm that there seems to be an issue with for-each loops. I should be able to release another update later today that fixes it.

**Comment by joshtynjala** [comment]
@yooni126 I just published v0.23.2. It fixes the issue with for-each. Sorry for the trouble!

**Comment by yooni126** [comment]
Yes, my issue has been fixed now 👍
